pandas/core/generic.py

```diff
@@ -7012,6 +7012,49 @@ def fillna(
         else:
             return result.__finalize__(self, method="fillna")
 
+    def downcast(self: NDFrameT) -> NDFrameT:
+        """Downcasts the columns to an appropriate dtype.
+
+        Possibly casts floats to integers. The dtype is inferred.
+
+        Returns
+        -------
+        {klass}
+            {klass} with the same shape and converted columns.
+
+        Notes
+        -----
+        The downcasting logic protects against truncating floats.
+        If the values don't fit into the specified dtype, the column is ignored.
+
+        Examples
+        --------
+        >>> df = pd.DataFrame({"foo": [1.0, 2.0], "bar": [1.5, 2.5], "baz": [3.0, 4.0]})
+        >>> df
+           foo  bar  baz
+        0  1.0  1.5  3.0
+        1  2.0  2.5  4.0
+
+        >>> result = df.downcast()
+        >>> result
+           foo  bar  baz
+        0    1  1.5    3
+        1    2  2.5    4
+
+        >>> result.dtypes
+        foo      int64
+        bar    float64
+        baz      int64
+        dtype: object
+        """
+        if using_copy_on_write():
+            result = self.copy(deep=False)
+        else:
+            result = self.copy(deep=True)
```

Comment:
i would have expected this to be handled within the Manager method.  am i wrong to be surprised?

Reply:
Doesn’t really matter, when we supported dict like inputs this was better, but can move it to the manager now